aiida/cmdline/commands/cmd_import.py

```diff
@@ -127,15 +136,64 @@ def cmd_import(archives, webpages, group, extras_mode_existing, extras_mode_new,
                 extras_mode_existing=ExtrasImportCode[extras_mode_existing].value,
                 extras_mode_new=extras_mode_new,
                 comment_mode=comment_mode)
-        except exceptions.IncompatibleArchiveVersionError as exception:
-            echo.echo_critical('{} cannot be imported: {}'.format(archive, exception))
+        except exceptions.IncompatibleArchiveVersionError:
+            echo.echo_warning('{} cannot be imported.'.format(archive))
+            if non_interactive:
+                migrate_archive = manual_migration
```

Comment:
I would have `no_migration=False` by default and make it the top-level logic.
If it is `True`, no need to even ask.
(you can print one line as the reason why no migration was performed)

Reply:
True. If one specifically specified `--manual-migration` / `--no-migration`, then there shouldn't be a prompt.

Reply:
I take it that you guys want the automatic migration to be the default behavior. Shouldn't a flag `--no-auto-migrate` then suffice? No need to make it hidden either. In scripts you can then specify `--non-interactive --no-auto-migrate`. I am trying to see if we really need the double flag and *if* we really need it, is it not better to keep the `--something/--no-something` dichotomy, i.e. `--auto-migrate/--no-auto-migrate`?

Reply:
> I take it that you guys want the automatic migration to be the default behavior.

Yes, see reasons and discussion conclusions [here](https://github.com/aiidateam/aiida_core/issues/2739#issuecomment-484109917).

>  ... is it not better to keep the `--something/--no-something` dichotomy, i.e. `--auto-migrate/--no-auto-migrate`?

I thought of this to begin with as well. But `--non-interactive` has other uses as well. There are several "Do you want to continue?" queries when an `Exception` is caught - for _any_ reason.
So I feel it is fine to have both, but they are definitely overlapping a bit, when it comes to the "Do you want to migrate?" query.

Reply:
What I meant with the double flag was not combination of non-interactive and the auto-migrate, but rather why `--no-auto-migrate/--auto-migrate` if we can simply have `--no-auto-migrate` only and by default do auto migration?

Reply:
I see no harm in having both the on- and off-switch.
I don't think it is really needed at the moment but it's one more line of code (or not even) and it could be used to explicitly force one behavior, even if the aiida developers at some point decide to change the default behavior.

I prefer `--no-migrate` over `--no-auto-migrate` since it's shorter and simpler (i.e. should I migrate or not?) but I'm fine with both.

Reply:
I will vote for `--no-migrate` as well, for the simplicity. And meaning is not lost, I think. 

But okay, I will make an on/off-switch: `--migration` / `--no-migration`

Reply:
Finally, do we agree on the following behaviour?:

| `migration` | `non_interactive` | Result CLI | Result script |
| ------------- | ---------------------- | ------------- | ---------------- |
| `True`* | `False`* | Query user, migrate | Migrate |
| `True`* | `True` | No query, migrate | Migrate |
| `False` | `False`* | No query, no migrate | No migrate |
| `False` | `True` | No query, no migrate | No migrate |

*Default

I do not know if there is a way to check whether a flag (`--migrate`) was specifically requested or not - since I could then implement that no query was performed, if it can be detected that it _was_ requested.

Reply:
I'm not quite sure what the difference between "result cli" and "result script" is here...
anyhow, the "result cli" is correct.

>  since I could then implement that no query was performed, if it can be detected that it was requested.

I think it's fine displaying a "no migration was performed" message whenever that is correct (whether a migration was requested or not). No need for more complex logic.

Reply:
i.e. if you literally mean "putting the same bash line into a script vs typing it on the terminal" - the behavior will be the same

Reply:
The difference was between calling `verdi import` in a terminal and calling it from, e.g. ansible.

Reply:
Great, done.